docs/draft-hfc-proposal.md

```diff
@@ -0,0 +1,70 @@
+The options for how we see to proceed.
+
+1) retain the status quo
+
+2) upstream block counting into the Ledger governance actions related to eras
+
+3) remove block counting, but keep double stability (ie permits the HFC interpreter to say Just Nothing Just)
+
+    3a) do not add HLQ mitigations
+    3b) add HLQ mitigations
+    3c) more exotic/subtle: change block counting to be a disjunct instead of a conjunct prevents Just -> Nothing (since that only arises when switching to a denser chain)
+
+4) remove block counting, remove double stability, and add HLQ mitigations
+
+Option 4 is elaborated below.
+
+-----
+
+A proposal for a simplified HFC, eg neither Counting Blocks nor requiring Double Stability.
+
+- Suppose a majority of stake is held by healthy caught-up honest nodes that are well-connected.
+  Thus those honest nodes exhibit Praos Common Prefix (CP) and Praos Chain Growth (CG).
+
+- In particular, CP ensures that the k+1st youngest block on the selection of any of those nodes is also already on the selection of all the other nodes and will remain there forever.
+  The block might not yet be the k+1st youngest on the other nodes' selections.
+  But none of these nodes will never need to switch away from its k+1st youngest block.
+
+- Accordingly, the ChainSync mini protocol client should disconnect from any upstream peer whose intersection with the node's selection's tip is more than k+1 blocks back.
+
+- Assume the ledger rules prevent any block from altering the leader schedule before a delay of one CG stability window and that, accordingly, ChainSync validates upstream peers' headers by forecasting the leadership schedule from the intersection of the peer's chain and the node's selection.
+
+    - In particular, this implies that the Ledger's governance for transitioning to the next era can also be forecasted by at least a stability window, since era transitions may affect the leader schedule.
+
+    - According to CG, this ensures the nodes will be able to validate at least k+1 blocks of an honest upstream peer's chain, which would suffice to incur the deepest intersection possibly allowed by CP.
+      Even so, the above rule is a reasonable behavior when the peer's chain violates CG: the number of validatable headers after the intersection is proportional to the severity of the CG violation.
```

Comment:
There *might* be an additional requirement that the era transition is known even more than a stability window ahead of the epoch boundary: https://github.com/IntersectMBO/ouroboros-consensus/issues/385#issuecomment-1747256702

The Slack thread mention there [has no reply so far](https://input-output-rnd.slack.com/archives/C02EF6ET4DC/p1696436827727239).